src/smt/theory_str_noodler/expr_cases.h

```diff
@@ -74,10 +71,45 @@ bool is_indexof_add(expr* e, expr* index_str, ast_manager& m, seq_util& m_util_s
  */
 bool is_indexof_at(expr * index_param, expr* index_str, ast_manager& m, seq_util& m_util_s);
 
+/**
+ * @brief Check if the expression @p e is of the form to_int(x) = num where num 
+ * is a number (or num = to_int(x)).
+ * 
+ * @param e Expression to be checked
+ * @param m Ast manager
+ * @param m_util_s string ast util
+ * @param m_util_a arith ast util
```

Comment:
I think generally we should skip these comments that do not say anything, you get the same information from the types of the arguments (I would remove this from other functions descriptions too that were added in this PR).

Reply:
It is fine to keep it there for the IDE. Moreover if a parameter comment is missing it might look like a mistake (doxygen would probably complain too). I am more thinking of some pattern class instead of just pure functions. In that case these common parameters could be stored as member variables.

Reply:
Hmm, you're right. I sometimes wonder if we shouldn't just put everything into `theory_str_noodler` class at this point (but keep it in separate files).